platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -681,28 +720,57 @@ export default class HangingProtocolService extends PubSubService {
    * simpler than recomputing the entire protocol.
    */
   public getComputedOptions(
-    options: Record<string, unknown>,
+    options: Record<string, unknown> | Array<Record<string, unknown>>,
     displaySetUIDs: string[]
-  ) {
-    const computed = { ...options };
-    let displaySets;
-    for (const key in computed) {
-      const value = computed[key];
-      if (!value) continue;
-      if (value.custom) {
-        if (!displaySets) {
-          displaySets = this.displaySets.filter(
-            displaySet =>
-              displaySetUIDs.indexOf(displaySet.displaySetInstanceUID) !== -1
+  ): any {
+    // Base case: if options is an array, map over the array and recursively call getComputedOptions
+    if (Array.isArray(options)) {
+      return options.map(option =>
+        this.getComputedOptions(option, displaySetUIDs)
+      );
+    }
+
+    if (typeof options === 'object' && options !== null) {
```

Comment:
You might return null if options is null, and then assume options is an array, so that you nest indentation 1 less, which makes it a bit easier to understand, eg:
if (Array...) ...

if( options===null ) return null;
// Assume options is object here - it has to be given the possibilities